src/util.ts

```diff
@@ -319,6 +329,40 @@ export function spawnChildProcess(
     });
 }
 
+// Excutes commands within backticks if they are marked as safe.
+export async function replaceCommands(x: string, safeCommands: string[] | undefined, opt: child_process.SpawnOptions): Promise<string> {
+    const regex = /`([^\s]+)\s(.*?)`/g;
+    let match = regex.exec(x);
+    while (match !== null) {
+        if (safeCommands?.some(c => c === match![1])) {
+            const output = await runCommand(`${match[1]} ${match[2]}`, opt);
+            x = x.replace(match[0], output);
+        } else {
+            logger.message(localize('unsafe.command', 'Commands must be marked as safe before they will be executed in the shell. Add {0} to the {1} setting if this command is safe to execute', match[1], `'makefile.safeCommands'`), 'Normal');
+            x = x.replace(match[0], `\`${match[1]} ${match[2]}\``);
+        }
+        match = regex.exec(x);
+    }
+    return x;
+}
+
+// Run a shell command and return the output.
+async function runCommand(x: string, opt: child_process.SpawnOptions): Promise<string> {
+    const child = child_process.spawn('/bin/bash', ['-c', `"${x}"`], opt);
```

Comment:
Here we need also the windows equivalent for "/bin/bash -c" which is "cmd -c". I'm doing the changes I am noticing as I review and test and will post a PR commit follow-up.

Reply:
My mistake, backtick does not apply on windows. A bailout to not run bin/bash on windows intead.

Reply:
`replaceCommands` should exit early on Windows.